File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function removeDuplicates(&$nums) {
        $l = 0;
        
        for($r = 1; $r <= count($nums); $r++){
            if($nums[$l] === $nums[$r]){
                continue;
            }
            
            $l++;
            $tmp = $nums[$r];
            $nums[$r] = $nums[$l];
            $nums[$l] = $tmp;
        }
        
        return $l++;
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    numsLen := len(nums)
    
    if numsLen <= 1 {
        return numsLen
    }
    
    l, r := 0, 1
    for r < numsLen {
        if nums[l] == nums[r] {
            r++
            continue
        }
        
        l++
        nums[l], nums[r] = nums[r], nums[l]
        r++
    }
    
    return l+1
}